models/users-model.js

```diff
@@ -0,0 +1,7 @@
+const db = require("../db/connection.js");
+
+exports.userByUsername = (username) => {
+    return db.query("SELECT * FROM users WHERE username = $1", [username]).then(({ rows }) => {
+        if (rows.length === 0) return Promise.reject({ status: 404, msg: "User not found" });
+    });
+};
```

Comment:
author in the comments table is a foreign key which relates to username in the users table. A psql error will be thrown if this is violated. It is not necessary to separately check if the user exists.